wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1046,15 +1046,24 @@ public final class MLSService: MLSServiceInterface {
         context.performAndWait {
             let conversations = ZMConversation.fetchConversationsWithPendingProposals(in: context)
 
-            result = conversations.compactMap { conversation in
+            for conversation in conversations {
                 guard
                     let groupID = conversation.mlsGroupID,
                     let timestamp = conversation.commitPendingProposalDate
                 else {
-                    return nil
+                    continue
                 }
 
-                return (groupID, timestamp)
+                result.append((groupID, timestamp))
+
+                // The pending proposal might be for the subconversation,
+                // so include it just in case.
+                if let subgroupID = subconverationGroupIDRepository.fetchSubconversationGroupID(
```

Comment:
so how can we be sure the pending proposal is for the subconversation?

from what I understand if it is for subconversation you also commit for the parent, no?

Reply:
We currently don't persist enough state to be able to determine at this point in time whether there exists a pending proposal for the parent, the sub conversation, or both. We could add this functionality, but I'm not sure that it is necessary.

We already don't exactly know which groups have pending proposals since only CC knows. We simply mark `ZMConversation`s as possibly having a pending proposal and eagerly try to commit them. If there are pending proposals, they will be committed, if there are none, then that's ok and nothing bad will happen.

> from what I understand if it is for subconversation you also commit for the parent, no?

The parent and subconversation have different group ids and I've found that we need to use the right id, otherwise CC returns some puzzling errors. In this case, I was trying to commit the pending proposals for the parent group, but CC had the pending proposal associated with the subgroup, so it was throwing "no pending proposals" which was confusing because I could clearly see we were getting it and it was valid.